locust/test/test_dispatch.py

```diff
@@ -3291,6 +3291,241 @@ class User3(User):
+                {"User1": 1, "User2": 1, "User3": 1, "User4": 1, "User5": 1},
+                {"User1": 13, "User2": 26, "User3": 39, "User4": 52, "User5": 1},
+                {"User1": 10, "User2": 10, "User3": 10, "User4": 10, "User5": 10},
+                {"User1": 10, "User2": 10, "User3": 10, "User4": 10, "User5": 10},
+            ],
+            user_classes=[User1, User2, User3, User4, User5],
+        )
+
+    def test_ramp_up_ramp_down_and_rump_up_again_fixed(self):
+        for weights, fixed_counts in [
+            [(1, 1, 1, 1, 1), (100, 100, 50, 50, 200)],
+            [(1, 1, 1, 1, 1), (100, 150, 50, 50, 0)],
+            [(1, 1, 1, 1, 1), (200, 100, 50, 0, 0)],
+            [(1, 1, 1, 1, 1), (200, 100, 0, 0, 0)],
+            [(1, 1, 1, 1, 1), (200, 0, 0, 0, 0)],
+            [(1, 1, 1, 1, 1), (0, 0, 0, 0, 0)],
+        ]:
+
+            u1_weight, u2_weight, u3_weight, u4_weight, u5_weight = weights
+            u1_fixed_count, u2_fixed_count, u3_fixed_count, u4_fixed_count, u5_fixed_count = fixed_counts
+
+            class User1(User):
+                weight = u1_weight
+                fixed_count = u1_fixed_count
+
+            class User2(User):
+                weight = u2_weight
+                fixed_count = u2_fixed_count
+
+            class User3(User):
+                weight = u3_weight
+                fixed_count = u3_fixed_count
+
+            class User4(User):
+                weight = u4_weight
+                fixed_count = u4_fixed_count
+
+            class User5(User):
+                weight = u5_weight
+                fixed_count = u5_fixed_count
+
+            target_user_counts = [sum(fixed_counts), sum(fixed_counts) + 100]
+            down_counts = [0, max(min(fixed_counts) - 1, 0)]
+            user_classes = [User1, User2, User3, User4, User5]
+
+            for worker_count in [3, 5, 9]:
+                workers = [WorkerNode(str(i + 1)) for i in range(worker_count)]
+                users_dispatcher = UsersDispatcher(worker_nodes=workers, user_classes=user_classes)
+
+                for down_to_count in down_counts:
+                    for target_user_count in target_user_counts:
+
+                        # Ramp-up to go to `target_user_count` #########
+
+                        users_dispatcher.new_dispatch(target_user_count=target_user_count, spawn_rate=1)
+                        users_dispatcher._wait_between_dispatch = 0
+
+                        list(users_dispatcher)
+
+                        for user_class in user_classes:
+                            if user_class.fixed_count:
+                                self.assertEqual(
+                                    users_dispatcher._get_user_current_count(user_class.__name__),
+                                    user_class.fixed_count,
+                                )
+
+                        # Ramp-down to go to `down_to_count`
+                        # and ensure what the fixed users was decreased too
```

Comment:
Unless I misunderstood the sentence, I think the "what" does not belong in that sentence:
```suggestion
                        # and ensure the fixed users was decreased too
```

Reply:
Removed